tests/test_rankings_storage.py

```
@@ -0,0 +1,257 @@
+from backend import rankings_storage
+from backend.elo import INITIAL_RATING
+
+
+@pytest.fixture
+def isolated_data_dir(tmp_path, monkeypatch):
+    """Point storage at a fresh temp dir for each test."""
+    monkeypatch.setattr(rankings_storage, "DATA_BASE_DIR", str(tmp_path))
+    return tmp_path
+
+
+def _stage2_round(label_to_model: dict[str, str], voter_rankings: list[tuple[str, list[str]]]):
```

Ruff (0.15.11)

[warning] Missing return type annotation for private function `_stage2_round`

(ANN202)